crypto/thr_id.c

```diff
@@ -199,6 +204,8 @@ void CRYPTO_THREADID_current(CRYPTO_THREADID *id)
     /* Else pick a backup */
 #if defined(OPENSSL_SYS_WIN32)
     CRYPTO_THREADID_set_numeric(id, (unsigned long)GetCurrentThreadId());
+#elif defined(__unix__) || defined(unix)
+    CRYPTO_THREADID_set_numeric(id, (unsigned long)pthread_self());
```

Comment:
It seems to be short enough on Linux/OS X. `&errno` on OS X, on the other hand, doesn't work.

Perhaps I should confine this to OS X?